lib/suspenders/adapters/heroku.rb

```diff
@@ -0,0 +1,50 @@
+module Suspenders
+  module Adapters
+    class Heroku
+      def initialize(app_builder)
+        @app_builder = app_builder
+      end
+
+      def set_heroku_remotes
+        remotes = <<-SHELL
+
+# Set up the staging and production apps.
+#{join_heroku_app('staging')}
+#{join_heroku_app('production')}
+      SHELL
+
+        @app_builder.append_file 'bin/setup', remotes
+      end
+
+      def join_heroku_app(environment)
+        heroku_app_name = heroku_app_name_for(environment)
+        <<-SHELL
+if heroku join --app #{heroku_app_name} &> /dev/null; then
+  git remote add #{environment} git@heroku.com:#{heroku_app_name}.git || true
+  printf 'You are a collaborator on the "#{heroku_app_name}" Heroku app\n'
+else
+  printf 'Ask for access to the "#{heroku_app_name}" Heroku app\n'
+fi
+        SHELL
+      end
+
+      private
+
+      def heroku_app_name_for(environment)
+        "#{@app_builder.app_name.dasherize}-#{environment}"
+      end
+
+      def run_heroku(command, environment)
+        path_addition = override_path_for_tests
+        @app_builder.run "#{path_addition} heroku #{command} --remote #{environment}"
+      end
+
+      def override_path_for_tests
+        if ENV['TESTING']
+          support_bin = File.join(ROOT_PATH, 'spec', 'fakes', 'bin')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.